src/repos.cpp

```diff
@@ -571,9 +571,10 @@ class TempDir {
 
 TempDir::TempDir()
 {
-    auto path = std::string { tmpnam(nullptr) };
-    std::filesystem::create_directory(m_path);
-    m_path = m_path;
+    char* temp = "tempdirXXXXXX";   
+    
+    auto path = std::string { mkdtemp(temp) };
```

Comment:
You cannot create it directly? `mkdtemp("tempdirXXXXXX")`? And is this correct? `tempdirXXXXXX`? It's ugly.